### ISSUE TITLE
[v2.9] _curses.error: curses function returned NULL

### ISSUE THREAD

**Issue by geekscrapy** [saulpw/visidata#1353]
**Small description**
_This seems to happen intermittently and I cant reliably repro..._

When I go to close a sheet with `quit-sheet`, sometimes I get the following error:

```
Traceback (most recent call last):
  File "/Users/geekscrapy/py3.7/lib/python3.7/site-packages/visidata/basesheet.py", line 167, in execCommand
    escaped = super().execCommand2(cmd, vdglobals=vdglobals)
  File "/Users/geekscrapy/py3.7/lib/python3.7/site-packages/visidata/basesheet.py", line 69, in execCommand2
    exec(code, vdglobals, LazyChainMap(vd, self))
  File "quit-sheet", line 1, in <module>
    import plugins.vpager
  File "/Users/geekscrapy/py3.7/lib/python3.7/site-packages/visidata/sheets.py", line 1068, in quit
    vs.confirmQuit('quit')
  File "/Users/geekscrapy/py3.7/lib/python3.7/site-packages/visidata/sheets.py", line 1077, in confirmQuit
    vd.confirm(f'{verb} modified sheet "{vs.name}?" ')
  File "/Users/geekscrapy/py3.7/lib/python3.7/site-packages/visidata/form.py", line 87, in confirm
    ret = FormCanvas(source=form).run(vd.scrFull)
  File "/Users/geekscrapy/py3.7/lib/python3.7/site-packages/visidata/form.py", line 55, in run
    self.scrForm = vd.scrFull.derwin(maxh+2, maxw+1, (h-maxh)//2-1, (w-maxw)//2-1)
_curses.error: curses function returned NULL
```

I can however do anything else (work with the sheet, `ctrl+i`, etc.) I just can't close.

**Expected result**
To close the sheet and push the next on the stack

**Additional context**
I see this occurring when I use `vd.run` from within another curses application (prompt-toolkit to be exact), which is probably the cause. Having said this though, I have a something niggling in my head telling me it happened when I used vd natively as well...

I'll add more context when it occurs again

**Comment by geekscrapy** [comment]
context:

Occurrance 1
- the sheet stack was not empty when trying to use `quit-sheet`
- cannot use `quit-all` either
- I can still manually delete the sheets from the `sheetsSheet`

Occurrence 2
- I just had a situation whereby I dont have any sheets in `sheetsSheet` when this issue occurred.
- I am calling `vd.run` multiple times within the parent curses app, and between `vd` sessions I am clearing with: 
```python
    vd.sheets = list()
    vd.statuses.clear()
    vd.allSheets.clear()
```

**Comment by saulpw** [comment]
This could happen if the sheet has been modified and the name of the sheet is too large to fit in the confirm window prompt.  I guess the confirm window should have a max width of ~80 and the sheet name should be truncated within it?  Or wrapped?

**Comment by geekscrapy** [comment]
Repro'd:

1. Sheet name was 180+ chars. I just dived a lot of rows
2. Added a capture-col of `(.)`
3. Error came up when I tried to close the sheet (**_after_** confirming with the dialog box)

I could not repro if I manually changed the sheet name in `SheetsSheet` though, it must be the original sheet name?

Also, just to confirm, for each session, I see the dialog box at least once before the error comes
